test/user.test.js

```diff
@@ -0,0 +1,100 @@
+const { connectDatabase, disconnectDatabase } = require("../utils/test-utils/db.util");
+
+const supertest = require('supertest')
+const app = require('../app');
+const request = supertest(app);
+
+const User = require('../db/schema/user');
+const userTestData = require('./user.test.data');
+
+describe('user routes', () =>{
+    beforeAll(() => {
+        connectDatabase();
+    });
+
+    beforeEach(async ()=>{
+        for(const userData of userTestData){
+            const newUser = new User({
+                uuid: userData.uuid,
+                userName: userData.userName,
+                password: userData.password
+            })
+            await newUser.save()
+        }
+    })
+    
+    afterEach(async ()=>{
+        await User.deleteMany()
+    })
+
+    afterAll(() => {
+        disconnectDatabase();
+    });
+    describe('GET route', () =>{
+        it('should get all users from database', async () =>{
+            const res = await request.get('/api/users');
+            expect(res.status).toBe(200)
+            expect(res.body.length).toBe(userTestData.length)
+            expect(res.body[0].uuid).toBe(userTestData[0].uuid)
+            expect(res.body[0].userName).toBe(userTestData[0].userName)
+            expect(res.body[0].password).toBe(userTestData[0].password)
+            expect(res.body[0].preferredPaymentMethod).toBe(userTestData[0].preferredPaymentMethod)
+        })
+
+    })
+    describe('POST route', () =>{
+        it('should add a new user to the database', async () =>{
+            const res = await request
+                .post('/api/users')
+                .send({
+                    uuid: '4',
+                    userName: 'testD',
+                    password: 'testPassD',
+                    preferredPaymentMethod: 'CASH'
+                })
+
+            expect(res.status).toBe(200)
+            expect(res.body.uuid).toBe('4')
```

Comment:
So this test checks for the response after we make the API request instead of checking whether we added the user to the database. Let's query for a record in the database and see if the record has the information we sent the post request to.

Reply:
okay will do